src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -783,3 +807,37 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
             "message": message
         }
     # endregion
+
+    # region - Patch Truncation
+    def __read_status_file_raw_data(self, file_path):
+        for i in range(0, Constants.MAX_FILE_OPERATION_RETRY_COUNT):
+            try:
+                with self.env_layer.file_system.open(file_path, 'r') as file_handle:
+                    status_file_data_raw = json.load(file_handle)[0]    # structure is array of 1
+            except Exception as error:
+                if i < Constants.MAX_FILE_OPERATION_RETRY_COUNT - 1:
+                    time.sleep(i + 1)
+                else:
+                    self.composite_logger.log_error("Unable to read status file (retries exhausted). Error: {0}.".format(repr(error)))
+                    raise
+        return status_file_data_raw
+
+    def __get_latest_status_complete_file(self, file_path):
+        """ Get the latest status complete file and remove other .complete.status files"""
+        list_of_files = glob.glob(file_path + '\\' + '*.complete.status')
+        latest_file = max(list_of_files,
+                          key=lambda x: (os.path.getmtime(x), int(re.search(r'(\d+)\.complete.status', x).group(1)), x))
+
+        for file in list_of_files:
+            if file != latest_file:
+                self.env_layer.file_system.delete_files_from_dir(file, '*.complete.status')
+
+        return latest_file
+
+    def __get_latest_file_kb_size(self, latest_file):
+        """ Get the file kb size in two decimal places of the latest status complete file"""
+        if latest_file is None:
+            raise Exception("Error: File is not found.")
+        file_kb_size = round(os.path.getsize(latest_file) / 1024, 2)
+        return file_kb_size
+    # endregion
```

Comment:
Files must always end with a new line. You'll get flagged in the IDE and in Github for this. (you can see it here).